TACC/Core-Portal
# task/FP-200: Trash Modal: Official Design

## Overview: ##
Redesigned trash process to remove intermediate modal, and added empty action when in .Trash to permanently delete files.

## Related Jira tickets: ##

* [FP-200](https://jira.tacc.utexas.edu/browse/FP-200)

## Summary of Changes: ##
- Replaced `TOGGLE_MODAL` dispatch with `DATA_FILES_TRASH` dispatch
- Added `trashInProgress` bool which checks trash status for files
- Added styling option, depending on `trashInProgress`, which disables interactions with file(s) being trashed.
- 'Trash' toolbar button becomes purple 'Empty' button when in .Trash
- Repurposed `DataFilesTrashModal` into `DataFilesEmptyModal` which allows user to confirm (with fair warning) deleting all files in .Trash

## Testing Steps: ##
1. _**Only test the empty action with files you don't care about, as they will be deleted.**_
2. Navigate to Data Files, then My Data (Work).
3. Select file(s) and trash them - confirm each selected row has greyed text, a spinner in place of the checkbox, and disabled interactions while trash action is in progress (see attached video for comparison).
4. Navigate to the .Trash folder - confirm the 'Trash' button now reads 'Empty' and is enabled.
5. Click the 'Empty' button - confirm the modal matches the attached modal picture.
6. Hit the 'Delete Files' button in the modal - confirm that modal disappears upon completion, notification reading '{n} file(s) deleted' appears on the bottom left of the window, that files listing in trash reads 'No files or folders to show.', and that the 'Empty' button is now disabled (reference attached video of empty process).

## UI Photos:

https://user-images.githubusercontent.com/43251554/145113626-602d0180-442f-493c-af18-208831e8ba8a.mov

<img width="1159" alt="Empty trash modal" src="https://user-images.githubusercontent.com/43251554/145113630-4b178134-0e80-4c66-96f0-89da3d3ca37c.png">

https://user-images.githubusercontent.com/43251554/145113646-34d33c5e-1719-4a31-a901-a905c5b53135.mov


## Notes: ##
